codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsSoaRecord.java

```diff
@@ -0,0 +1,121 @@
+     *                 <ul>
+     *                     <li>{@link #CLASS_IN}</li>
+     *                     <li>{@link #CLASS_CSNET}</li>
+     *                     <li>{@link #CLASS_CHAOS}</li>
+     *                     <li>{@link #CLASS_HESIOD}</li>
+     *                     <li>{@link #CLASS_NONE}</li>
+     *                     <li>{@link #CLASS_ANY}</li>
+     *                 </ul>
+     * @param timeToLive the TTL value of the record
+     * @param primaryNameServer the name server that was the original or primary source of data for this zone.
+     * @param responsibleAuthorityMailbox the mailbox of the person responsible for this zone.
+     * @param serialNumber the version number of the original copy of the zone.
+     * @param refreshInterval the time interval before the zone should be refreshed.
+     * @param retryInterval the time interval that should elapse before a failed refresh should be retried.
+     * @param expireLimit the time value that specifies the upper limit on the time interval
+     * @param minimumTTL the minimum TTL field that should be exported with any RR from this zone.
+     */
+    public DefaultDnsSoaRecord(
+            String name, int dnsClass, long timeToLive,
+            String primaryNameServer, String responsibleAuthorityMailbox,
+            int serialNumber, int refreshInterval, int retryInterval, int expireLimit, int minimumTTL) {
+        super(name, DnsRecordType.SOA, dnsClass, timeToLive);
+        this.primaryNameServer = checkNotNull(primaryNameServer, "primaryNameServer");
+        this.responsibleAuthorityMailbox = checkNotNull(responsibleAuthorityMailbox, "responsibleAuthorityMailbox");
+        this.serialNumber = serialNumber;
+        this.refreshInterval = refreshInterval;
+        this.retryInterval = retryInterval;
+        this.expireLimit = expireLimit;
+        this.minimumTTL = minimumTTL;
+    }
+
+    @Override
+    public String primaryNameServer() { return primaryNameServer; }
+
+    @Override
+    public String responsibleAuthorityMailbox() { return responsibleAuthorityMailbox; }
+
+    @Override
+    public int serialNumber() { return serialNumber; }
+
+    @Override
+    public int refreshInterval() { return refreshInterval; }
+
+    @Override
+    public int retryInterval() { return retryInterval; }
+
+    @Override
+    public int expireLimit() { return expireLimit; }
+
+    @Override
+    public int minimumTTL() { return minimumTTL; }
+
+    @Override
+    public String toString() {
+        final StringBuilder buf = new StringBuilder(64).append(StringUtil.simpleClassName(this)).append('(');
+        final DnsRecordType type = type();
+        buf.append(name().isEmpty()? "<root>" : name())
+                .append(' ')
+                .append(timeToLive())
+                .append(' ');
+
+        DnsMessageUtil.appendRecordClass(buf, dnsClass())
+                .append(' ')
+                .append(type.name());
+
+        buf.append(' ')
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) StringBuffer (or StringBuilder).append is called 2 consecutive times with literal Strings. Use a single append with a single combined String. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AConsecutiveLiteralAppends)